### PR TITLE
Added scripts ica-ega-encryptor and ica-ega-uploader

### PR DESCRIPTION
**Pull request by alexiswl**
Usages are as below:

## Encryptor

Given a bam path, encrypt bam file and index into an output destination path

```
ica-ega-encryptor \
  --gds-bam-path gds://development/temp/alexis/apgi-umccrise-test/apgi2137/output/20230424_095916/ICGC_0002/work/APGI_2137__APGI_2137_tumour/oncoviruses/work/detect_viral_reference/host_unmapped_or_mate_unmapped_to_gdc.bam \
  --output-path gds://development/temp/alexis/ega_encryption_test_output/APGI_2137/ \
  --logs-path gds://development/temp/alexis/ega_encryption_test_logs/APGI_2137
```

## Uploader

Given a directory of files, upload the directory to EGA

```
ica-ega-uploader \
  --gds-path gds://development/temp/alexis/ega_encryption_test_output/APGI_2137/ \
  --logs-path gds://development/temp/alexis/ega_upload_logs/ \
  --ega-dest-directory /path/to/cohort/ \
  --ega-username foo-user \
  --ega-password foo-password
```

### CODE REVIEW

**Review by brainstorm — APPROVED** [review]
Legend!

**Comment by alexiswl** [comment]
Few teething issues here. See https://github.com/umccr-illumina/stratus/issues/193 have literally tried every possible combination of things. Didn't help that I hadn't set the command correctly for a quite a few iterations.  

tl;dr ICA TES is pretty tightly wrapped and I don't think allows the 33001 port.  If we don't want to pay for egress, I think we refactor the encrypt command to push instead to ICAv2 (a non BYOB bucket) and then upload from bench
